File: BackendService/BackendService/src/main/java/reap/backend/Entities/Restaurant.java
package reap.backend.Entities;


import jakarta.persistence.*;

import java.util.List;

@Entity
public class Restaurant {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private int id;
    private String name;
    private String phoneNumber;
    private String emailAddress;
    private String address;
    private String restaurantType;
    private String website;
    private boolean delivery;


    @OneToMany(mappedBy = "restaurant")
    private List<Reservation> reservations;

    public Restaurant() {}

    public Restaurant(String name, String phoneNumber, String emailAddress, String address, String restaurantType, String website, boolean delivery) {
        this.name = name;
        this.phoneNumber = phoneNumber;
        this.emailAddress = emailAddress;
        this.address = address;
        this.restaurantType = restaurantType;
        this.website = website;
        this.delivery = delivery;
    }

    public int getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public String getPhoneNumber() {
        return phoneNumber;
    }

    public String getEmailAddress() {
        return emailAddress;
    }

    public String getAddress() {
        return address;
    }

    public String getRestaurantType() {
        return restaurantType;
    }

    public String getWebsite() {
        return website;
    }

    public boolean isDelivery() {
        return delivery;
    }

    public List<Reservation> getReservations() {
        return reservations;
    }

    public void setName(String name) {
        this.name = name;
    }

    public void setPhoneNumber(String phoneNumber) {
        this.phoneNumber = phoneNumber;
    }

    public void setEmailAddress(String emailAddress) {
        this.emailAddress = emailAddress;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public void setRestaurantType(String restaurantType) {
        this.restaurantType = restaurantType;
    }

    public void setWebsite(String website) {
        this.website = website;
    }

    public void setDelivery(boolean delivery) {
        this.delivery = delivery;
    }
}

File: Gui/GUI/lib/pojos/Restaurant.dart
class Restaurant {
  late String name;
  late String phoneNumber;
  late String emailAddress;
  late String address;
  late String restaurantType;
  late String website;
  late bool delivery;
  late String selfHref;

  Restaurant({
    required this.name,
    required this.phoneNumber,
    required this.emailAddress,
    required this.address,
    required this.restaurantType,
    required this.website,
    required this.delivery,
    required this.selfHref
  });

  Restaurant.fromJson(Map<String, dynamic> json) {
    name = json['name'];
    phoneNumber = json['phoneNumber'];
    emailAddress = json['emailAddress'];
    address = json['address'];
    restaurantType = json['restaurantType'];
    website = json['website'];
    delivery = json['delivery'];
    selfHref = json['_links']['self']['href'];
  }
}